src/main/java/io/cdap/plugin/gcp/bigquery/util/BigQueryUtil.java

```diff
@@ -579,6 +580,27 @@ public static void validateTable(String table, String tablePropertyName, Failure
     match(table, tablePropertyName, TABLE_PATTERN, collector, errorMessage);
   }
 
+  /**
+   * Validates allowed GCS Upload Request Chunk Size.
+   *
+   * @param chunkSize provided chunk size
+   * @param chunkSizePropertyName GCS chunk size name property
+   * @param collector failure collector
+   */
+  public static void validateGCSChunkSize(String chunkSize, String chunkSizePropertyName, FailureCollector collector) {
+    String errorMessage1 = String.format("Value must be a multiple of %s", MediaHttpUploader.MINIMUM_CHUNK_SIZE);
+    String errorMessage2 = "Error parsing the provided value";
+    if (!Strings.isNullOrEmpty(chunkSize)) {
+      try {
+        if (Integer.parseInt(chunkSize) % MediaHttpUploader.MINIMUM_CHUNK_SIZE != 0) {
+          collector.addFailure(errorMessage1, null).withConfigProperty(chunkSizePropertyName);
+        }
+      } catch (NumberFormatException e) {
+        collector.addFailure(errorMessage2, null).withConfigProperty(chunkSizePropertyName);
```

Comment:
You can use e.getMessage() as error message and add corrective action.

Reply:
fixed